src/typing.ml

```diff
@@ -635,6 +636,60 @@ let rec val_parse_core_type ns cty =
       ((ty_of_core ns ct1, lbl) :: args, res)
   | _ -> ([], ty_of_core ns cty)
 
+let rec check_old ns wr t =
+  let array_get = ns_find_ls ns [ "Gospelstdlib"; "Array"; "get" ] in
+  let rec check t =
+    match t.t_node with
+    | Tvar vs ->
+        if
+          not
+            (List.exists
+               (function { t_node = Tvar vs' } -> vs = vs' | _ -> false)
+               wr)
+        then Some vs.vs_name.id_str
+        else None
```

Comment:
I don't think that would be enough. For instance, you would reject the following:
```ocaml
val f : 'a array -> unit
(*@ f x 
    modifies x 
    ensures let y = x in old y.(0) = 0
```
You need a more subtle data flow analysis to know which variables are actually modified when you write `modifies x`.